term-project.el

```diff
@@ -0,0 +1,155 @@
+  (term-manager-purge-dead-buffers term-project-term-manager)
+  (term-manager-get-all-buffers term-project-term-manager))
+
+(defun term-project-select-existing ()
+  "Select `project-term' buffer in the current project."
+  (completing-read "Select a term buffer: "
+                   (mapcar 'buffer-name
+                           (term-project-get-all-buffers))))
+
+;;;###autoload
+(defun term-project-switch-to ()
+  "Switch to an existing `term-project' buffer using `completing-read'."
+  (interactive)
+  (term-manager-display-buffer (term-project-select-existing)))
+
+;;;###autoload
+(defun term-project-forward ()
+  "Switch forward to the next `term-project' `ansi-term' buffer.
+Make a new one if none exists."
+  (interactive)
+  (term-project-switch :symbol (term-project-maybe-intern (project-root (project-current)))))
+
+;;;###autoload
+(defun term-project-backward ()
+  "Switch backward to the next `term-project' `ansi-term' buffer.
+Make a new one if none exists."
+  (interactive)
+  (term-project-switch :delta -1 :symbol (term-project-maybe-intern (project-root (project-current)))))
+
+;;;###autoload
+(cl-defun term-project-create-new (&optional (directory (project-root (project-current))))
+  "Make a new `ansi-term' buffer for DIRECTORY.
+If directory is nil, use the current project project"
+  (interactive)
+  (when (stringp directory) (setq directory (term-project-maybe-intern directory)))
+  (term-manager-display-buffer
+   (term-manager-build-term term-project-term-manager directory)))
+
+;;;###autoload
+(defun term-project-default-directory-forward ()
+  "Switch to the next `term-project' `ansi-term' buffer for `defualt-directory'."
+  (interactive)
+  (term-project-switch :symbol default-directory))
+
+;;;###autoload
+(defun term-project-default-directory-backward ()
+  "Switch to the prev `term-project' `ansi-term' buffer for `defualt-directory'."
+  (interactive)
+  (term-project-switch :delta -1 :symbol default-directory))
+
+;;;###autoload
+(defun term-project-default-directory-create-new ()
+  "Make a new `ansi-term' buffer in `default-directory'."
+  (interactive)
+  (let ((directory (if (stringp default-directory) (term-project-maybe-intern default-directory))))
+    (term-project-create-new directory)))
+
+;;;###autoload
+(defun term-project-global-forward ()
+  "Switch forward to the next `term-project' `ansi-term' buffer.
+Make a new one if none exists."
+  (interactive)
+  (term-project-global-switch :delta 1))
+
+;;;###autoload
+(defun term-project-global-backward ()
+  "Switch backward to the next `term-project' `ansi-term' buffer.
+Make a new one if none exists."
+  (interactive)
+  (term-project-global-switch :delta -1))
+
+;;;###autoload
+(defun term-project-global-create-new ()
+  "Make a new `ansi-term' buffer in `term-project-global-directory'."
+  (interactive)
+  (term-project-create-new term-project-global-directory))
+
+(provide 'term-project)
+;;; term-project.el ends here
```

Comment:
a lot of this is really boilerplate. Perhaps we should write a macro that can generate all of these functions.